packages/teleport/src/services/websession/storeWebSession.ts

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import history from 'teleport/services/history';
+
+import {
+  KeysEnum,
+  EventKeys,
+  BroadcastChannelMessage,
+  BearerToken,
+} from './types';
+
+const isTest = process.env.NODE_ENV === 'test';
+
+export class StoreWebSession {
+  bcBroadcaster: BroadcastChannel;
+  bcReceiver: BroadcastChannel;
+
+  constructor() {
+    this.initBroadcastChannel();
```

Comment:
In interest of removing the test specific code, the constructor can be expanded to take a configuration object which would contain the broadcast channel.